tests/core/test_docs.py

```diff
@@ -76,9 +76,33 @@
 def test_docs(function):
     docstrings = function.__doc__
     arguments = inspect.signature(function).parameters
+
     assert all(
         [
             _PARAMETER_MAPPING.get(argument, "Parameters") in docstrings
             for argument in arguments
         ]
     )
+
+    assert all([argument in docstrings for argument in arguments])
+
+
+def test_docs_duplicity():
+    @default_docs
+    def dummy_function(store):
+        """
+        This is a dummy_function
+
+        Parameters
+        -----------
+        store: str
+            This is an argument
+
+        """
+
+    resultant_docstrings = str(dummy_function.__doc__)
+    assert "This is a dummy_function" in resultant_docstrings
+    assert "This is an argument" in resultant_docstrings
+    assert (
+        "Factory dummy_function producing a KeyValueStore" not in resultant_docstrings
```

Comment:
Should be `Factory function producing a KeyValueStore.`. Also, can you add a short comment/docstring here to explain what you're doing?